code/datums/elements/slapcrafting.dm

```diff
@@ -14,35 +15,29 @@
  * It will check the area near the user for the rest of the ingredients and tools.
  * *
 **/
-/datum/component/slapcrafting/Initialize(
-		slapcraft_recipes = null,
-	)
+/datum/element/slapcrafting/Attach(datum/target, slapcraft_recipes = null)
+	..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
 
-	if(!isitem(parent))
-		return COMPONENT_INCOMPATIBLE
+	var/obj/item/target_item = target
 
-	var/obj/item/parent_item = parent
+	if((target_item.item_flags & ABSTRACT) || (target_item.item_flags & DROPDEL))
+		return ELEMENT_NOTRANSFER
 
-	if((parent_item.item_flags & ABSTRACT) || (parent_item.item_flags & DROPDEL))
-		return COMPONENT_NOTRANSFER
+	RegisterSignal(target, COMSIG_ATOM_ATTACKBY, PROC_REF(attempt_slapcraft))
+	RegisterSignal(target, COMSIG_ATOM_EXAMINE, PROC_REF(get_examine_info))
+	RegisterSignal(target, COMSIG_ATOM_EXAMINE_MORE, PROC_REF(get_examine_more_info))
+	RegisterSignal(target, COMSIG_TOPIC, PROC_REF(topic_handler))
 
-	RegisterSignal(parent, COMSIG_ATOM_ATTACKBY, PROC_REF(attempt_slapcraft))
-	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, PROC_REF(get_examine_info))
-	RegisterSignal(parent, COMSIG_ATOM_EXAMINE_MORE, PROC_REF(get_examine_more_info))
-	RegisterSignal(parent, COMSIG_TOPIC, PROC_REF(topic_handler))
+	// You need to turn lists into string lists on attach() for bespoke elements or they'll all be still separate instances
+	src.slapcraft_recipes = string_list(slapcraft_recipes)
```

Comment:
```suggestion
	src.slapcraft_recipes = slapcraft_recipes
```

Yes, but not here. It's to be done before `AddElement()` (or `RemoveElement()`) is even called, since `GetElement()` is what creates the instances if it doesn't find one with the exact same args.

Basically, it's to be done case by case when the unit test tells you to. Otherwise leave it as a static list.